File: src/com/esprit/veltun/model/Abonnement.java

package com.esprit.veltun.model;

import com.esprit.veltun.model.base.BaseEntity;

import java.sql.Date;

public class Abonnement extends BaseEntity {
    private int Id_ab;
    private float Prix_ab;
    private String Type_ab;
    private int Id_offre;
    private String Duree;

    public Date getDateFin() {
        return dateFin;
    }

    public void setDateFin(Date dateFin) {
        this.dateFin = dateFin;
    }

    public Date getDateDebut() {
        return dateDebut;
    }

    public void setDateDebut(Date dateDebut) {
        this.dateDebut = dateDebut;
    }

    private Date dateDebut;
    private Date dateFin;
    public String getDuree() {
        return this.Duree;
    }

    public void setDuree(String Duree) {
        this.Duree = Duree;
    }

    public String getType_ab() {
        return this.Type_ab;
    }

    public void setType_ab(String Type_ab) {
        this.Type_ab = Type_ab;
    }


    public int getId_offre() {
        return this.Id_offre;
    }

    public void setId_offre(int Id_offre) {
        this.Id_offre = Id_offre;
    }

    public Abonnement(int i) {
        this.Id_ab = this.Id_ab;
    }

    public Abonnement() {
    }
    public Abonnement( int Id_ab,String Type_ab, Date dateDebut, Date dateFin, float Prix_ab, int Id_offre) {
        this.Id_ab = Id_ab;
        this.Type_ab = Type_ab;
        //this.Duree = Duree;
        this.dateDebut=dateDebut;
        this.dateFin=dateFin;
        this.Prix_ab = Prix_ab;
        this.Id_offre = Id_offre;

    }
    public Abonnement( int Id_ab,String Type_ab, String Duree, float Prix_ab, int Id_offre) {
        this.Id_ab = Id_ab;
        this.Type_ab = Type_ab;
        this.Duree = Duree;
        this.Prix_ab = Prix_ab;
        this.Id_offre = Id_offre;

    }
    public Abonnement( String Type_ab, String Duree, float Prix_ab, int Id_offre) {
        this.Type_ab = Type_ab;
        this.Duree = Duree;
        this.Prix_ab = Prix_ab;
        this.Id_offre = Id_offre;

    }



    public int getId_ab() {
        return this.Id_ab;
    }

    public void setId_ab(int Id_ab) {
        this.Id_ab = Id_ab;
    }

    public float getPrix_ab() {
        return this.Prix_ab;
    }

    public void setPrix_ab(float Prix_ab) {
        this.Prix_ab = Prix_ab;
    }

    public String toString() {
        return "abonnement{Id_ab=" + this.Id_ab + ", Type_ab=" + this.Type_ab + ", date_debut=" + this.dateDebut + ", date_fin=" + this.dateFin + ", Prix_ab=" + this.Prix_ab + '}';
    }
}

File: webProject/src/Entity/Abonnement.php
<?php

namespace App\Entity;
use App\Repository\AbonnementRepository;
use Doctrine\ORM\Mapping as ORM;
use Symfony\Component\Validator\Constraints as Assert;


#[ORM\Entity(repositoryClass: AbonnementRepository::class)]
class Abonnement
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $idAb = null;

    #[ORM\Column(length: 30)]
    private ?string $typeAb = null;

    #[ORM\Column(type: 'datetime')]
    private ?\DateTime $dateDebut = null;

    #[ORM\Column(type: 'datetime')]
    #[Assert\GreaterThan(propertyPath: 'dateDebut')]
    private ?\DateTime $dateFin = null;

    #[ORM\Column(type: 'float')]
    #[Assert\NotBlank(message: 'price non valid')]
    private ?float $prixAb = null;

    #[ORM\Column(type: 'integer')]
    #[ORM\ManyToOne(inversedBy: 'Offre')]
    #[Assert\NotNull]
    #[Assert\Entity(
        entityClass: Offre::class,
        message: 'Invalid Offre'
    )]
    private ?int $idOffre;

    public function getIdAb(): ?int
    {
        return $this->idAb;
    }

    public function setIdAb(?int $idAb): void
    {
        $this->idAb = $idAb;
    }

    public function getTypeAb(): ?string
    {
        return $this->typeAb;
    }

    public function setTypeAb(?string $typeAb): void
    {
        if (!in_array($typeAb, ['Standard', 'Premium', 'Premium+'])) {
            throw new \InvalidArgumentException('Invalid subscription type.');
        }
        $this->typeAb = $typeAb;
    }

    public function getDateDebut(): ?\DateTime
    {
        return $this->dateDebut;
    }

    public function setDateDebut(?\DateTime $dateDebut): void
    {
        $this->dateDebut = $dateDebut;
    }

    public function getDateFin(): ?\DateTime
    {
        return $this->dateFin;
    }

    public function setDateFin(?\DateTime $dateFin): void
    {
        $this->dateFin = $dateFin;
    }

    public function getPrixAb(): ?float
    {
        return $this->prixAb;
    }

    public function setPrixAb(?float $prixAb): void
    {
        if ($prixAb !== null && !is_float($prixAb)) {
            throw new \InvalidArgumentException('Price must be a float.');
        }
        $this->prixAb = $prixAb;
    }

    public function getIdOffre(): ?int
    {
        return $this->idOffre;
    }

    public function setIdOffre(?int $idOffre): void
    {
        $this->idOffre = $idOffre;
    }

}
